include/openthread/platform/mdns_socket.h

```diff
@@ -154,6 +158,47 @@ extern void otPlatMdnsHandleReceive(otInstance                  *aInstance,
                                     bool                         aIsUnicast,
                                     const otPlatMdnsAddressInfo *aAddress);
 
+/**
+ * Callback to notify OpenThread mDNS module of host address changes.
+ *
+ * When `otPlatMdnsSetListeningEnabled()` enables mDNS listening on an @p aInfraIfIndex, the platform MUST monitor and
+ * report ALL IPv4 and IPv6 addresses assigned to this network interface.
+ *
+ * When mDNS is enabled:
+ * - The platform MUST retrieve ALL currently assigned IPv4 and IPv6 addresses on the specified interface.
+ * - For each retrieved address, the platform MUST call `otPlatMdnsHandleHostAddressEvent()`.
+ * - The IPv4 addresses are represented using IPv4-mapped IPv6 format.
+ *
+ * Ongoing monitoring (while enabled):
+ * - The platform MUST continuously monitor the specified interface for address changes.
+ * - If any addresses are added or removed, the platform MUST call this callback for each affected address, indicating
+ *   the change (addition or removal using @p aAdded).
+ *
+ *  When mDNS is disabled:
+ * - The platform MUST cease monitoring for address changes on the interface.
+ * - The platform does NOT need to explicitly signal the removal of addresses upon disable. The OpenThread stack
+ *   automatically clears its internal address list.
+ * - If address monitoring is re-enabled later, the platform MUST repeat the "enable" steps again, retrieving and
+ *   reporting ALL current addresses.
+ *
+ * The OpenThread stack maintains an internal list of host addresses. It updates this list automatically upon receiving
+ * calls to `otPlatMdnsHandleHostAddressEvent()`.
+ * - OpenThread's mDNS implementation uses a short guard time before taking action (e.g., announcing new addresses).
```

Comment:
Is it worth calling out what the guard time is? Is it mentioned elsewhere?

Reply:
Added here that it is about `(4 msec)`. 
Updated other changes in the new pushed commit. Thanks.